### PR TITLE
[luci/export] Revise SerializedData for extended Buffer

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise SerializedData for extended Buffer support.

### CODE REVIEW

**Comment by seanshpark** [comment]
- for #13678
- from #13615

File: compiler/luci/export/src/SerializedData.h [hunk 1]
```diff
@@ -128,6 +128,16 @@ struct SerializedModelData final
   // This is used for removing buffers with same values
   std::map<luci::CircleConst *, uint32_t> _cached_buffer_id;
 
+  // flag to use extended Buffer mode for file size > 2G
+  bool _ext_buffer = false;
+  // flag to indicate flatbuffer area got size > 2G
+  bool _require_ext_buffer = false;
```
Comment (jinevening) [inline compiler/luci/export/src/SerializedData.h:134]: May I ask what is the difference of these two? Can they be different, e.g., one is true, one is false?

Comment (seanshpark) [reply compiler/luci/export/src/SerializedData.h:134]: assumption) we only know the file size > 2G while generating flatbuffers data.
in the first pass, `_require_ext_buffer` will be set to `true` if data > 2G and fail the first pass. 
if first pass fails and `_require_ext_buffer` is `true`, we set `_ext_buffer` to `true` and run the second pass.
while generation, if `_ext_buffer` is set, we do the extended Buffer mode of flatbuffers.

Comment (seanshpark) [reply compiler/luci/export/src/SerializedData.h:134]: you may see this implementation in draft https://github.com/Samsung/ONE/pull/13615/files#diff-4cc4ad0dc5cdf533fdb5e85d1963a994d2826143f04b80c8a3175a8762fc79db

Comment (jinevening) [reply compiler/luci/export/src/SerializedData.h:134]: Thanks for explanation. I just wondered if we could remove `_ext_buffer`. It's difficult to distinguish them just by looking at the comment.

**Review by jinevening — APPROVED** [review]
LGTM

**Pull request merged**